### DISCUSSION TITLE
Adding CD index metric

### DISCUSSION THREAD

**Discussion by evgepab** [networkx/networkx#6629] [Ideas]
Hello!

Me and @eirinikafourou were wondering if you would consider helpful/interesting adding the cd index metric for the graphs in NetworkX.

**Comment by MridulS** [comment]
Hi, could you give some information about this or any citations? I couldn't find much on google scholar https://scholar.google.com/scholar?hl=en&as_sdt=0%2C47&q=cd+index+metric

Thanks!

**Reply by evgepab** [reply]
I read about CD index from here: http://russellfunk.org/cdindex/ and the specific paper can be found here: http://russellfunk.org/cdindex/static/papers/funk_ms_2017.pdf . Hope this is helpful!

**Reply by evgepab** [reply]
If you'd like any more information regarding this, don't hesitate to let us know!

**Comment by dschult** [comment]
It looks like there is a [recent paper in Nature](https://www.nature.com/articles/s41586-022-05543-x) about the CD index too. And the field of time dependent network analysis is finally starting to be developed. We would be interested in  having a function (or more if appropriate) to compute the CD index contributed to NetworkX. You can create a PR for adding a module such as `networkx/algorithms/time_dependent.py` with tests in `networkx/algorithms/tests/test_time_dependent.py`.  You should read through the `CONTRIBUTING.md` file and other documentation for nx developers. Take a look at similar modules in the algorithms sub-package and try to mimic the docs and tests. Don't worry about having it be perfect for starting the PR. We can work with you over git and you just push to your fork and it updates the PR. 

Thanks!

**Reply by eirinikafourou** [reply]
Great, thank you for the feedback!! We will start working on this immediately!
